src/cljs/dato/lib/core.cljs

```diff
@@ -40,31 +41,26 @@
+(defmethod ws-message-handler :default
+  [dato-ch msg]
+  ;; TODO replace the dato-ch handler with this
+  (put! dato-ch msg))
+
+(defn new-dato [dato-host dato-port dato-path app-db cs-schema]
   (let [cast-bootstrapped? (atom false)
         ws-ch              (async/chan)
         dato-ch            (async/chan)
+        ;; XXX: can we get rid of this?
         ss-cast!           #(put! ws-ch %)
-        pending-rpc        (atom {})
-        ws                 (ws/build dato-host dato-path {:on-open    (fn [event] (js/console.log ":on-open :" event))
-                                                          :on-close   (fn [event] (js/console.log ":on-close :" event))
-                                                          :on-message (fn [event]
-                                                                        (let [fr (js/FileReader. (.-data event))]
-                                                                          (aset fr "onloadend" (fn [e]
-                                                                                                 (let [data (transit/read transit-reader (.. e -target -result))]
-                                                                                                   (if-let [cb-id (:server/rpc-id data)]
-                                                                                                     (let [cb-or-ch (get @pending-rpc cb-id)]
-                                                                                                       (swap! pending-rpc dissoc cb-id)
-                                                                                                       ;; Should we just pass the whole payload through, or just the data/results?
-                                                                                                       (cond
-                                                                                                         (chan? cb-or-ch) (put! cb-or-ch (:data data))
-                                                                                                         (fn? cb-or-ch)   (cb-or-ch (:data data))
-                                                                                                         :else            (ss-cast! (:data data))))
-                                                                                                     (ss-cast! data)))))
-                                                                          (.readAsText fr (.-data event))))
-                                                          :on-error   (fn [event] (js/console.log ":on-error :" event))})
+        ;; XXX: can we get rid of this?
+        tal-state          (tal/init {:port dato-port
+                                      :transit-reader transit-reader
+                                      :transit-writer transit-writer})
+        _                  (tal/start-message-consumer tal-state (fn [msg]
```

Comment:
I moved it into talaria's init function, as :message-handler, so that everything is in one place.